src/network/networkd-link.c

```diff
@@ -583,6 +587,65 @@ int link_get(Manager *m, int ifindex, Link **ret) {
         return 0;
 }
 
+int link_get_by_name(const Manager *m, const char *ifname, Link **ret) {
+        Link *link;
+
+        assert(m);
+        assert(ifname);
+        assert(ret);
+
+        link = hashmap_get(m->links_by_name, ifname);
+        if (!link)
+                return -ENODEV;
+
+        *ret = link;
+
+        return 0;
+}
+
+int link_update_by_name(Link *link, bool update_ifname, bool update_alternative_names) {
+        char **n;
+        int r;
+
+        assert(link);
+
+        r = hashmap_ensure_allocated(&link->manager->links_by_name, &string_hash_ops);
+        if (r < 0)
+                return r;
+
+        if (update_ifname)  {
+                r = hashmap_put(link->manager->links_by_name, link->ifname, link);
+                if (r < 0 && r != -EEXIST)
+                        return r;
+        }
+
+        if (update_alternative_names) {
+                STRV_FOREACH(n, link->alternative_names) {
+                        r = hashmap_put(link->manager->links_by_name, *n, link);
+                        if (r < 0 && r != -EEXIST)
+                                return r;
+                }
+        }
+
+        return 0;
+}
+
+int link_remove_by_name(Link *link, bool update_ifname, bool update_alternative_names) {
+        char **n;
+
+        assert(link);
+
+        if (update_alternative_names) {
+                STRV_FOREACH(n, link->alternative_names)
+                        hashmap_remove_value(link->manager->links_by_name, *n, link);
+        }
+
+        if (update_ifname)
+                hashmap_remove_value(link->manager->links_by_name, link->ifname, link);
+
+        return 0;
+}
```

Comment:
maybe have two distinct functions link_remove_by_altnames() and link_remove_by_ifname() here, and then call either one or the other, or both. the boolean thing is a bit weird